examples/ping.rs

```diff
@@ -15,12 +15,47 @@ use smoltcp::time::{Duration, Instant};
 use smoltcp::phy::Device;
 use smoltcp::phy::wait as phy_wait;
 use smoltcp::wire::{EthernetAddress, IpAddress, IpCidr,
+                    Ipv6Address, Icmpv6Repr, Icmpv6Packet,
                     Ipv4Address, Icmpv4Repr, Icmpv4Packet};
 use smoltcp::iface::{NeighborCache, EthernetInterfaceBuilder};
 use smoltcp::socket::{SocketSet, IcmpSocket, IcmpSocketBuffer, IcmpPacketMetadata, IcmpEndpoint};
 use std::collections::HashMap;
 use byteorder::{ByteOrder, NetworkEndian};
 
+macro_rules! send_icmp_ping {
+    ( $repr_type:ident, $packet_type:ident, $ident:expr, $seq_no:expr, $echo_payload:expr,
+      $socket:expr, $remote_addr:expr, $device_caps:expr ) => {{
```

Comment:
`device_caps` isn't used and what is the purpose of the double `{`?

Reply:
I forgot to remove `device_caps` after an intermediate version.

The first `{` is part of the `macro_rules!` syntax, the second one means it's a block.

Reply:
I forgot to remove `device_caps` after an intermediate version.

The first `{` is part of the `macro_rules!` syntax, the second one means it's a block.

Reply:
Ah right, because the body is essentially a rvalue.